storage/block.go

```diff
@@ -43,7 +44,8 @@ type dbBlock struct {
 	opts     m3db.DatabaseOptions
 	start    time.Time
 	encoder  m3db.Encoder
-	stream   m3db.SegmentReader
+	segment  *m3db.Segment
```

Comment:
I went back and forth on this, initially started with a struct but felt we allocated a large number of blocks in the server, which used quite a bit of memory, so I switched over to using a pointer instead.

Maybe I'll get rid of the alloc functions to compensate for the space overhead of having a struct inside.